cadc-rest/src/main/java/ca/nrc/cadc/rest/RestAction.java

```diff
@@ -193,6 +193,28 @@ protected void initState() {
         }
     }
 
+
+    /**
+     * Check if the caller can read the resource.
```

Comment:
This is not a permission check so has nothing to do with the specific caller. This is checking the service state to see if any read should go ahead. Please clarify the javadocs and also add an `@throws IllegalStateException` (it's a RuntimeException, so not requried but good documentation for implementers of subclasses.

also: the checkWritable javadocs
otherwise, code is fine

Reply:
Updated.